hazelcast/src/main/java/com/hazelcast/spi/impl/sequence/CallIdSequenceWithBackpressureFailFast.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi.impl.sequence;
+
+import com.hazelcast.core.HazelcastOverloadException;
+
+import java.util.concurrent.atomic.AtomicLongArray;
+
+import static com.hazelcast.nio.Bits.CACHE_LINE_LENGTH;
+import static com.hazelcast.nio.Bits.LONG_SIZE_IN_BYTES;
+
+/**
+ * A {@link CallIdSequence} that provides backpressure by taking
+ * the number of in-flight operations into account when before creating a new call-id.
+ * <p>
+ * It is possible to temporarily create more concurrent invocations than the declared capacity due to:
+ * <ul>
+ *     <li>system operations</li>
+ *     <li>the racy nature of checking if space is available and getting the next sequence. </li>
+ * </ul>
+ * The latter cause is not a problem since the capacity is exceeded temporarily and it isn't sustainable.
+ * So perhaps there are a few threads that at the same time see that the there is space and do a next.
+ */
+public final class CallIdSequenceWithBackpressureFailFast implements CallIdSequence {
```

Comment:
This callIdSequence doesn't provide back pressure. So Perhaps FailFastCallIdSequence is a better name.

Also there is now a lot of code duplication between this class and the CallIdSequenceWithBackpressure; so probably good to pull out shared superclass.

Reply:
There is no unit tests for this class.

Reply:
OK, Changed name to FailFastCallIdSequence. Combined the common methods at FailFastCallIdSequence and made the CallIdSequenceWithBackpressure extend from FailFastCallIdSequence. Added unit test FailFastCallIdSequenceTest. The ClientMaxAllowedInvocationTest also tests using the FailFastCallIdSequence.